test/ziggurat/messaging/consumer_test.clj

```diff
@@ -292,7 +305,41 @@
+        proto-message              (proto/->bytes (proto/create proto-class message))
+        message-payload            {:message proto-message :topic-entity topic-entity :retry-count 3}]
+    (testing "should return deserialized message-payload if serialized using protobuf"
+      (let [proto-serialized-message-payload (zmd/serialize-to-message-payload-proto message-payload)
+            converted-message-payload        (consumer/convert-and-ack-message nil {:delivery-tag 1} proto-serialized-message-payload false "default")]
+        (is (= (bytes-to-str converted-message-payload) (bytes-to-str message-payload)))))
+    (testing "should return deserialized message-payload with topic-entity as the keyword "
+      (let [proto-serialized-message-payload (zmd/serialize-to-message-payload-proto message-payload)
+            converted-message-payload        (consumer/convert-and-ack-message nil {:delivery-tag 1} proto-serialized-message-payload false "default")]
+        (is (keyword? (:topic-entity converted-message-payload)))
+        (is (= (bytes-to-str converted-message-payload) (bytes-to-str message-payload)))))
+    (testing "should return deserialized message-payload with the message as a byte array "
+      (let [proto-serialized-message-payload (zmd/serialize-to-message-payload-proto message-payload)
+            converted-message-payload        (consumer/convert-and-ack-message nil {:delivery-tag 1} proto-serialized-message-payload false "default")]
+        (is (= "class [B" (str (type (:message converted-message-payload)))))
+        (is (= (bytes-to-str converted-message-payload) (bytes-to-str message-payload)))))
+    (testing "should return a map and not ziggurat.message_payload/->MessagePayload if original message was a ziggurat.message_payload/->MessagePayload and serialized using nippy"
+      (let [nippy-message-payload         (assoc (zmp/->MessagePayload proto-message topic-entity) :retry-count 3)
+            nippy-serialized-message-payload (nippy/freeze nippy-message-payload)
+            converted-message-payload        (consumer/convert-and-ack-message nil {:delivery-tag 1} nippy-serialized-message-payload false "default")]
+        (is (= (bytes-to-str converted-message-payload) (bytes-to-str message-payload)))))
+    (testing "should return a map and not ziggurat.mapper/->MessagePayload if original message was a ziggurat.mapper/->MessagePayload and serialized using nippy"
+      (let [nippy-message-payload         (assoc (mpr/->MessagePayload proto-message topic-entity) :retry-count 3)
+            nippy-serialized-message-payload (nippy/freeze nippy-message-payload)
+            converted-message-payload        (consumer/convert-and-ack-message nil {:delivery-tag 1} nippy-serialized-message-payload false "default")]
+        (is (= (bytes-to-str converted-message-payload) (bytes-to-str message-payload)))))
+    (testing "should return a clojure map if serialized using nippy"
```

Comment:
I had only written this test. But, the  title is wrong.

Please change it to: "should return a nil if the message is serialized using neither proto or nippy"